src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt

```diff
@@ -38,10 +37,13 @@ import kotlin.reflect.KClass
 
 open class DexMethodCountPlugin: Plugin<Project> {
     companion object {
-        var sdkLocation: File? = SdkResolver.resolve(null)
-        const val VERSION_3_ZERO_FIELD: String = "com.android.builder.Version" // <= 3.0
-        const val VERSION_3_ONE_FIELD: String = "com.android.builder.model.Version" // > 3.1
-        const val AGP_VERSION_FIELD: String = "ANDROID_GRADLE_PLUGIN_VERSION"
+        var sdkLocation: File? = null
+        private const val VERSION_3_ZERO_FIELD: String = "com.android.builder.Version" // <= 3.0
+        private const val VERSION_3_ONE_FIELD: String = "com.android.builder.model.Version" // > 3.1
+        private const val AGP_VERSION_FIELD: String = "ANDROID_GRADLE_PLUGIN_VERSION"
+        private const val AGP_VERSION_3: String = "3.0.0"
+        private const val ANDROID_EXTENSION_NAME = "android"
+        private const val SDK_DIRECTORY_METHOD = "getSdkDirectory"
```

Comment:
This is the first I've ever heard of this method - has it always been part of the `android` extension?  What gaps exist between it and `SdkResolver`?

Reply:
@benjamin-bader This has been around since at least Android Gradle Plugin 1.5, see the documentation here: http://google.github.io/android-gradle-dsl/1.5/com.android.build.gradle.AppExtension.html#com.android.build.gradle.AppExtension:sdkDirectory.

The only gap I see is if they suddenly remove this but the have had this API since 1.5 and we are on 3.1.0 now.